test_unstructured/partition/test_msg.py

```diff
@@ -156,102 +132,49 @@ def test_partition_msg_from_file_exclude_metadata():
-
-    mocked_last_modification_date = "2029-07-05T09:24:28"
-
-    attachment_elements = partition_text(
-        filename=attachment_filename,
-        metadata_filename=attachment_filename,
-        metadata_last_modified=mocked_last_modification_date,
-    )
-    expected_metadata = attachment_elements[0].metadata
-    expected_metadata.file_directory = None
-    expected_metadata.attached_to_filename = file_path
-
-    elements = partition_msg(
-        filename=file_path,
-        attachment_partitioner=partition_text,
-        process_attachments=True,
-        metadata_last_modified=mocked_last_modification_date,
-    )
-
-    # This test does not need to validate if hierarchy is working
-    # Patch to nullify parent_id
-    expected_metadata.parent_id = None
-    elements[-1].metadata.parent_id = None
-
-    assert elements[0].text.startswith("Hello!")
-    for element in elements[:-1]:
-        assert element.metadata.filename == "fake-email-attachment.msg"
-        assert element.metadata.subject == "Fake email with attachment"
-    assert elements[-1].text == "Hey this is a fake attachment!"
-    assert elements[-1].metadata == expected_metadata
-
-
-def test_partition_msg_can_process_min_max_wtih_attachments(tmp_path: pathlib.Path):
-    file_path = example_doc_path("fake-email-attachment.msg")
-    tmp_dir_path = str(tmp_path)
-    extract_msg_attachment_info(filename=file_path, output_dir=tmp_dir_path)
-    attachment_filename = str(tmp_path / str(ATTACH_EXPECTED_OUTPUT[0]["filename"]))
-
-    attachment_elements = partition_text(
-        filename=attachment_filename,
-        metadata_filename=attachment_filename,
-        min_partition=6,
-        max_partition=12,
-    )
-
+def test_partition_msg_can_process_attachments():
     elements = partition_msg(
-        filename=file_path,
-        attachment_partitioner=partition_text,
-        process_attachments=True,
-        min_partition=6,
-        max_partition=12,
-    )
-
-    assert elements[0].text.startswith("Hello!")
-    assert elements[-1].text == attachment_elements[-1].text
-    assert elements[-2].text == attachment_elements[-2].text
-    for element in elements:
-        if element.metadata.attached_to_filename is not None:
-            assert len(element.text) <= 12
-            assert len(element.text) >= 6
-
-
-def test_partition_msg_raises_with_no_partitioner():
-    with pytest.raises(ValueError):
-        partition_msg(example_doc_path("fake-email-attachment.msg"), process_attachments=True)
-
-
-def test_partition_msg_metadata_date_from_header(mocker: MockFixture):
-    expected_last_modification_date = "2022-12-16T17:04:16-05:00"
-
-    mocker.patch(
-        "unstructured.partition.msg.get_last_modified_date",
-        return_value=None,
-    )
-    mocker.patch(
-        "unstructured.partition.msg.get_last_modified_date_from_file",
-        return_value=None,
+        example_doc_path("fake-email-attachment.msg"), process_attachments=True
```

Comment:
Not related to this PR per se, but I don't think there's a way to flip on attachment processing in the API right now. We may want to enable that in the API or make `process_attachments=True` the default.

Reply:
Hmm, good point. Not much sense in having it if we can't turn it on :)

I'd be inclined to default it to `True` then, in this PR, and then see about following up with a set of PRs to add an API parameter to the SDKs, API, ingest, etc.

Another idea that occurs to me is we could tie this to `strategy`, which is already a supported parameter, and have `fast` skip attachments and anything else (`auto`, `hi_res`, `ocr_only` I suppose too) partition any attachments.

What do you think of that idea?

Reply:
Since this is already a big PR, let's keep the default the same for this one and we can work a separate PR to update the default for that parameter. One quirk is that `strategy` could also apply to how the attachment gets processed, and if you said `process_attachments=True` if `strategy="hi_res"`, you can never process PDF attachments with `"fast"`. We can sort that out in a follow on PR though and get input from the API crew (cc @rakepat @amanda103 )

Reply:
Hmm, yes, good point, I hadn't thought of that.

On a related note, we currently don't pass along _any_ parameters to the attachment partitioner, so there's no way to specify a non-default option.

That gap presented itself as I was doing the rewrite and I added a `.partitioning_kwargs` property to `MsgPartitionerOptions` that is passed along to `partition()` for each attachment. It's currently empty but we can populate it with whatever we think makes sense. https://github.com/Unstructured-IO/unstructured/pull/3142/files?show-viewed-files=true&file-filters%5B%5D=#diff-ff46992888c6719e6d8f804d3ed6c8c1b736be31528a359c7987191b904bf012R143-R148

We have the second PR in this series (new_msg.py -> msg.py) that I'll put up as soon as this merges, so we can add some pass-through arguments there if we want. Either way it's simple to do here, maybe not so much from the API or whatever.

`partition_msg()` really only takes `process_attachments` which no other file-format uses, and remaining "generic" parameters (e.g. `date_from_file_object`, etc.) wouldn't apply to attachments. So maybe we can just pass along `kwargs`.

But yeah, maybe we need to add a `partition_attachments: bool = False` parameter to the API, SDK, ingest, etc.